server/src/main/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorService.java

```diff
@@ -369,6 +370,13 @@ List<UserAction.Definition> diagnoseUnassignedShardRouting(ShardRouting shardRou
             case DECIDERS_NO:
                 actions.addAll(explainAllocationsAndDiagnoseDeciders(shardRouting, state));
                 break;
+            case NO_ATTEMPT:
+                if (UnassignedInfo.Reason.NODE_LEFT == shardRouting.unassignedInfo().getReason()) {
+                    // If shardRouting is delayed and no actions are returned then we CAN allocate the shard,
```

Comment:
I found this as a corner case late in the testing. The problem is that the only way to ensure that a delayed allocation will eventually be assigned is to diagnose the shard and find that there are no actions returned (because all deciders ended up with YES answers). If we are not returning details, the diagnosis doesn't happen, and thus we can't check this case on all health calls yet. I wasn't sure how best to approach that case since the index is technically red, but the best action is to get the node back online or wait and let Elasticsearch handle it.

I think it might be reasonable to add a conditional here with a user action that simply says "Wait for the shard to be reassigned", but that might be better to include in a follow up PR after some more discussion and with more testing.